ledger/participant-integration-api/src/main/scala/platform/store/backend/common/EventStorageBackendTemplate.scala

```diff
@@ -310,76 +340,64 @@ abstract class EventStorageBackendTemplate(
         )
     }
 
-  private def rawTreeEventParser(
-      allQueryingParties: Set[Int]
-  ): RowParser[EventsTable.Entry[Raw.TreeEvent]] =
-    createdTreeEventParser(allQueryingParties) | exercisedTreeEventParser(allQueryingParties)
-
-  private val selectColumnsForTransactionTreeCreate = Seq(
-    "event_offset",
-    "transaction_id",
-    "node_index",
-    "event_sequential_id",
-    "event_id",
-    "contract_id",
-    "ledger_effective_time",
-    "template_id",
-    "workflow_id",
-    "create_argument",
-    "create_argument_compression",
-    "create_signatories",
-    "create_observers",
-    "create_agreement_text",
-    "create_key_value",
-    "create_key_value_compression",
-    "NULL as exercise_choice",
-    "NULL as exercise_argument",
-    "NULL as exercise_argument_compression",
-    "NULL as exercise_result",
-    "NULL as exercise_result_compression",
-    "NULL as exercise_actors",
-    "NULL as exercise_child_event_ids",
-    "submitters",
-  ).mkString(", ")
-
-  private val selectColumnsForTransactionTreeExercise = Seq(
-    "event_offset",
-    "transaction_id",
-    "node_index",
-    "event_sequential_id",
-    "event_id",
-    "contract_id",
-    "ledger_effective_time",
-    "template_id",
-    "workflow_id",
-    "NULL as create_argument",
-    "NULL as create_argument_compression",
-    "NULL as create_signatories",
-    "NULL as create_observers",
-    "NULL as create_agreement_text",
-    "create_key_value",
-    "create_key_value_compression",
-    "exercise_choice",
-    "exercise_argument",
-    "exercise_argument_compression",
-    "exercise_result",
-    "exercise_result_compression",
-    "exercise_actors",
-    "exercise_child_event_ids",
-    "submitters",
-  ).mkString(", ")
+  private case class EventPartition[T](
+      tableName: String,
+      selectColumns: String,
+      parser: Set[Int] => RowParser[T],
+  );
+
+  private val partitionFlatCreate = EventPartition[EventsTable.Entry[Raw.FlatEvent]](
+    tableName = "participant_events_create",
+    selectColumns = selectColumnsForFlatTransactionsCreate,
+    parser = createdFlatEventParser _,
+  )
+  private val partitionFlatConsumingExercise = EventPartition[EventsTable.Entry[Raw.FlatEvent]](
+    tableName = "participant_events_consuming_exercise",
+    selectColumns = selectColumnsForFlatTransactionsExercise,
+    parser = archivedFlatEventParser _,
```

Comment:
It was mostly a mechanical transformation of existing code. "Archived" events are actually "consuming exercise" events, which have the exact same table structure as "non-consuming exercise" events (that's why the parser can be reused). I assume it was called `archivedFlatEventParser` because it returns a `Raw.FlatEvent.Archived` object. There is no `Raw.FlatEvent.Exercised`. 

Now I wonder why `def flatTransaction` is looking at non-consuming exercise events however. The [original code](https://github.com/digital-asset/daml/blob/a9fd5497ef5af24a682510e64482b5f1c5a85298/ledger/participant-integration-api/src/main/scala/platform/store/backend/common/EventStorageBackendTemplate.scala#L495) only filtered out divulgence events, even though non-consuming exercise events should not be part of flat transactions.